compact/range.go

```diff
@@ -111,6 +110,29 @@ func (r *Range) AppendRange(other *Range, visitor VisitFn) error {
 	return r.appendImpl(other.end, other.hashes[0], other.hashes[1:], visitor)
 }
 
+// Merge extends the compact range by merging in the other compact range from
+// the right. It uses the tree hasher to calculate hashes of newly created
+// nodes, and reports them through the visitor function (if non-nil). The other
+// range must begin between the current range's begin and end.
+//
+// Warning: This method modifies both this and the other Range.
```

Comment:
It seems unnecessary to modify the other range, no?  Or at least this feels like it could easily have some awkward and unintended side effects.  Why not implement it in a more "pure" way?